bot/exts/moderation/infraction/_utils.py

```diff
@@ -269,3 +272,43 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    def __init__(
+        self,
+        *,
+        allowed_users: t.Sequence[int],
+        allowed_roles: t.Sequence[int],
+        timeout: t.Optional[float] = 180,
+        message: t.Optional[Message] = None
+    ) -> None:
+        super().__init__(
+            allowed_users=allowed_users,
+            allowed_roles=allowed_roles,
+            timeout=timeout,
+            message=message
```

Comment:
What does this do? You don't pass `message` in any scenario and I can't see the purpose of it.
I also don't see it being used

Reply:
> What does this do? You don't pass `message` in any scenario and I can't see the purpose of it. I also don't see it being used

I see the point of `self.message` now, however you don't need to pass it into the `super` class
```suggestion
```

Reply:
You're absolutely right. I was debating this myself, but I decided to just leave it and have a 1:1 between arguments of the parent class (`ViewWithUserAndRoleCheck`) and the subclass (`StaffBanConfirmationView`). It really can't hurt and it's there in case we ever need it if we decide to re-use the same class elsewhere (we might for other commands, like cleanban, warnings, mutes, etc)

Reply:
The only reason I've included it is because setting the `self.value` to `None` by default. If I understand how subclassing works in Python, once you include an `__init__` of your own in a subclass, it overrides the parent's `__init__`, yes?

Reply:
There doesn't seem to be a pattern available to emit a yes/no response from the interaction. Only a success/timeout.

Unless we want to wrap the entire infraction step in a method, structure it as a callback, and hand it to the buttons for this interaction; I'm not seeing a way around toting the boolean flag `value` around. To complicate that hand-off, it looks like the result of the infraction call needs to be returned again from [the calling method here](https://github.com/python-discord/bot/blob/63c44a47c3e8bf15acf5d6babc2466664da6d63b/bot/exts/moderation/infraction/infractions.py#L528)

Reply:
I realise I never responded to this. What I meant was we don't need this init function specifically, as there's no need to set `confirmed` to None.

when we wait for this view we can do `timed_out = await confirmation_view.wait()` then check `if timed_out` first, meaning that will cover the case where `confirmed` isn't set.